File: benchmarks/c/sieve.c
#include <stdio.h>
#include <stdint.h>
#include <string.h>

int main(void) {
    int64_t limit = 100000;
    char sieve[100001];
    memset(sieve, 0, sizeof(sieve));
    for (int64_t i = 2; i * i <= limit; i++) {
        if (sieve[i] == 0) {
            for (int64_t j = i * i; j <= limit; j += i) {
                sieve[j] = 1;
            }
        }
    }
    int64_t count = 0;
    for (int64_t i = 2; i <= limit; i++) {
        if (sieve[i] == 0) count++;
    }
    printf("%ld\n", count);
    return 0;
}

File: benchmarks/python/sieve.py
limit = 1000000
sieve = [False] * (limit + 1)
i = 2
while i * i <= limit:
    if not sieve[i]:
        j = i * i
        while j <= limit:
            sieve[j] = True
            j += i
    i += 1

count = 0
for i in range(2, limit + 1):
    if not sieve[i]:
        count += 1

print(count)
